__tests__/main.test.ts

```diff
@@ -0,0 +1,248 @@
+import { run } from "../src/main"
+
+// Mock our edits interface to simply succeed
+jest.mock("../src/edits", () => {
+    const originalModule = jest.requireActual("../src/edits")
+
+    return {
+        __esModule: true,
+        ...originalModule,
+        runUpload: Promise.resolve()
+    }
+})
+
+// We need to mock setFailed so we can actually capture issues instead of setting the process exit code
+jest.mock("@actions/core", () => {
+    const originalModule = jest.requireActual("@actions/core")
+
+    return {
+        __esModule: true,
+        ...originalModule,
+        setFailed: jest.fn((message: string | Error) => {
+            if (message instanceof Error) {
+                throw message
+            }
+            throw Error(message)
+        })
+    }
+})
+
+function initInputs(
+    serviceAccountJson: string | undefined,
+    serviceAccountJsonRaw: string | undefined,
+    packageName: string | undefined,
+    releaseFile: string | undefined,
+    releaseFiles: string | undefined,
+    releaseName: string | undefined,
+    track: string | undefined,
+    inAppUpdatePriority: string | undefined,
+    userFraction: string | undefined,
+    status: string | undefined,
+    whatsNewDir: string | undefined,
+    mappingFile: string | undefined,
+    debugSymbols: string | undefined,
+    changesNotSentForReview: string | undefined,
+    existingEditId: string | undefined,
+) {
+    if (serviceAccountJson) {
```

Comment:
Somebody please tell me there's a better way to handle this

Reply:
There is a package, it just took a bit of digging to find
https://github.com/cardinalby/github-action-ts-run-api